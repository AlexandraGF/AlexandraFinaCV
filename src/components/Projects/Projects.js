import React, { Component } from 'react';
import plusImage from '../../plus.png';
import minusImage from '../../minus.png';
import './Projects.scss';

class Projects extends Component {
    constructor(props) {
        super(props);
        this.state = {clicked: false};
    
        this.handleClick = this.handleClick.bind(this);
      }

      handleClick() {
        this.setState(state => ({
          clicked: !state.clicked
        }))
    };

    render() {
        const projectsList = [ 
            {id: 1, name: 'Puzzle Game (React)', link: 'https://puzzlegamereact.herokuapp.com'},
            {id: 2, name: 'Funny Story', link: 'https://alexandragf.github.io/funnyStory-JavaScript/'},  
            {id: 3, name: 'Tic Tac Toe', link: 'https://alexandragf.github.io/ticTacToeJs/'}, 
        ]; 

        return (
        <div> 
            <div className="projectHeading" onClick={this.handleClick}>
                <div className="projectTitle">Personal Projects</div>
                <div className="projectIconImage">{!this.state.clicked ? (<img src={plusImage} />) : (<img src={minusImage} />)}</div>
            </div>
            <div className="projectsBody">{this.state.clicked ? (
                <div>
                    {projectsList.map(function(item, i){
                        return <a target="_blank" href={item.link}><div className="listItemProject" key={i}>{item.name}</div></a>
                    })} 
                </div>) : "" }
            </div>
        </div>
        );
    }
}

export default Projects;
