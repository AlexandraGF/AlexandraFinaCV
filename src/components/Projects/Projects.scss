@import '../../variables.scss';

.projectHeading {
    font-family: 'Allura', cursive;
    font-size: 30px;
    margin: 7% 7% 0 11%;
    padding-left: 6%;
    box-shadow: $boxShadow;
    border-radius: $borderRadius;
    display: inline-block;
    width: 80%;
}

.projectTitle {
    width: 80%;
}

.projectIconImage {
    width: 8%;
}

.projectTitle, .projectIconImage {
    flex-wrap: wrap;
    display: inline-flex;
}

.projectHeading:hover {
    transform: $hoverSize;
    cursor: $cursorPointer;
}

.projectBody {
    box-shadow: 5px 10px 12px #a2c3c3;
    margin-top: 7%;
    margin-left: 7%;
    font-family: 'K2D', sans-serif;
}

.listItemProject {
    margin: 7% 26% 5% 32%;
    padding: 1% 1% 1%;
    box-shadow: $boxShadow;
}

.listItemProject:hover {
    transform: $hoverSize;
}
