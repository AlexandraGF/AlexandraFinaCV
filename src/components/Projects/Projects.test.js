import React from 'react';
import ReactDOM from 'react-dom';
import Enzyme, {shallow, mount} from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';

Enzyme.configure({ adapter: new Adapter() });

import Projects from './Projects';

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<Projects />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('renders text', () => {
    const wrapper = mount(<Projects />);
    expect(wrapper.text()).toContain("Personal Projects");
    });
    
    it('renders Projects Component', () => {
    const tree = renderer.create(
    <Projects />
    ).toJSON();
    
    expect(tree).toMatchSnapshot();
    })
