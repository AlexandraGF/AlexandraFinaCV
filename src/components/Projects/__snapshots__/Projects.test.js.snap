// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders App Component 1`] = `
<div>
  <div
    className="projectHeading"
    onClick={[Function]}
  >
    <div
      className="projectTitle"
    >
      Personal Projects
    </div>
    <div
      className="projectIconImage"
    >
      <img
        src="plus.png"
      />
    </div>
  </div>
  <div
    className="projectsBody"
  >
    
  </div>
</div>
`;

exports[`renders Projects Component 1`] = `
<div>
  <div
    className="projectHeading"
    onClick={[Function]}
  >
    <div
      className="projectTitle"
    >
      Personal Projects
    </div>
    <div
      className="projectIconImage"
    >
      <img
        src="plus.png"
      />
    </div>
  </div>
  <div
    className="projectsBody"
  >
    
  </div>
</div>
`;
